addons/wiki/migrations/0007_migrate_node_wiki_pages.py

```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.9 on 2018-02-20 17:30
+from __future__ import unicode_literals
+
+import logging
+import progressbar
+from django.db import migrations
+from osf.utils.migrations import disable_auto_now_add_fields
+from django.contrib.contenttypes.models import ContentType
+from bulk_update.helper import bulk_update
+from addons.wiki.utils import to_mongo_key
+logger = logging.getLogger(__name__)
+
+def repoint_id(guid, old_wiki_page):
+    old_wiki_page._id = guid._id
+    old_wiki_page.save()
+    return
+
+def reverse_func(state, schema):
+    """
+    This will not restore all NodeWikiPages, where NodeWikiPage was later renamed.
+    """
+    NodeWikiPage = state.get_model('addons_wiki', 'nodewikipage')
+    AbstractNode = state.get_model('osf', 'abstractnode')
+    WikiVersion = state.get_model('addons_wiki', 'wikiversion')
+    WikiPage = state.get_model('addons_wiki', 'wiki_page')
+    for node in AbstractNode.objects.exclude(wiki_pages_versions={}):
+        for wiki in node.wikis.filter(deleted__isnull=True):
+            old_wiki_page_ids = node.wiki_pages_versions[to_mongo_key(wiki.page_name)]
+            old_wiki_pages = NodeWikiPage.objects.filter(node_id=node.id, page_name=wiki.page_name).order_by('version')
+            for index, old_wiki_page_id in enumerate(old_wiki_page_ids):
+                old_wiki_page = old_wiki_pages[index]
+                guid = migrate_guid_referent(old_wiki_page_id, old_wiki_page)
+                repoint_id(guid, old_wiki_page)
+            move_comment_target(wiki, old_wiki_page)
+            update_comments_viewed_timestamp(node, wiki, old_wiki_page)
+    WikiVersion.objects.all().delete()
+    WikiPage.objects.all().delete()
+
+def get_guid(state, osf_model):
+    Guid = state.get_model('osf', 'guid')
+    return Guid.objects.get(object_id=osf_model.id, content_type_id=ContentType.objects.get_for_model(osf_model).id)
+
+def move_comment_target(state, current_target, desired_target):
+    """Move the comment's target from the current target to the desired target"""
+    Comment = state.get_model('osf', 'comment')
+    current_target_guid = get_guid(state, current_target)
+    desired_target_guid = get_guid(state, desired_target)
```

Comment:
We're querying for the contenttypes of WikiPage and NodeWikiPage here. Again, it'd be ideal if we only did that once.